src/main/java/spoon/support/reflect/reference/CtArrayTypeReferenceImpl.java

```diff
@@ -85,7 +86,7 @@ public String getQualifiedName() {
 	public Class<T> getActualClass() {
 		Class<?> c = getComponentType().getActualClass();
 		if (c == null) {
-			return null;
+			throw new SpoonClassNotFoundException("you should never call getActualClass", null);
```

Comment:
Could the exception contain the qualified name of the class which is not found? .. it might improve debugging